src/main/java/jcifs/smb1/netbios/NbtAddress.java

```diff
@@ -0,0 +1,880 @@
+                return true;
+            }
+        }
+        return false;
+    }
+    static InetAddress switchWINS() {
+        nbnsIndex = (nbnsIndex + 1) < NBNS.length ? nbnsIndex + 1 : 0;
+        return NBNS.length == 0 ? null : NBNS[nbnsIndex];
+    }
+
+    Name hostName;
+    int address, nodeType;
+    boolean groupName,
+        isBeingDeleted,
+        isInConflict,
+        isActive,
+        isPermanent,
+        isDataFromNodeStatus;
+    byte[] macAddress;
+    String calledName;
+
+    NbtAddress( Name hostName, int address, boolean groupName, int nodeType ) {
+        this.hostName = hostName;
+        this.address = address;
+        this.groupName = groupName;
+        this.nodeType = nodeType;
+    }
+
+    NbtAddress( Name hostName,
+                int address,
+                boolean groupName,
+                int nodeType,
+                boolean isBeingDeleted,
+                boolean isInConflict,
+                boolean isActive,
+                boolean isPermanent,
+                byte[] macAddress ) {
+
+/* The NodeStatusResponse.readNodeNameArray method may also set this
+ * information. These two places where node status data is populated should
+ * be consistent. Be carefull!
+ */
+        this.hostName = hostName;
+        this.address = address;
+        this.groupName = groupName;
+        this.nodeType = nodeType;
+        this.isBeingDeleted = isBeingDeleted;
+        this.isInConflict = isInConflict;
+        this.isActive = isActive;
+        this.isPermanent = isPermanent;
+        this.macAddress = macAddress;
+        isDataFromNodeStatus = true;
+    }
+
+/* Guess next called name to try for session establishment. These
+ * methods are used by the smb package.
+ */
+
+    public String firstCalledName() {
+
+        calledName = hostName.name;
+
+        if( Character.isDigit( calledName.charAt( 0 ))) {
+            int i, len, dots;
+            char[] data;
+
+            i = dots = 0;                    /* quick IP address validation */
+            len = calledName.length();
+            data = calledName.toCharArray();
+            while( i < len && Character.isDigit( data[i++] )) {
+                if( i == len && dots == 3 ) {
+                    // probably an IP address
+                    calledName = SMBSERVER_NAME;
+                    break;
+                }
+                if( i < len && data[i] == '.' ) {
+                    dots++;
+                    i++;
+                }
+            }
+        } else {
+            switch (hostName.hexCode) {
+                case 0x1B:
+                case 0x1C:
+                case 0x1D:
+                    calledName = SMBSERVER_NAME;
+            }
+        }
+
+        return calledName;
+    }
+    public String nextCalledName() {
+
+        if( calledName == hostName.name ) {
+            calledName = SMBSERVER_NAME;
+        } else if( calledName == SMBSERVER_NAME ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.